Modules/Segmentation/ConnectedComponents/include/itkRelabelComponentImageFilter.h

```diff
@@ -207,7 +207,7 @@ class ITK_TEMPLATE_EXPORT RelabelComponentImageFilter : public InPlaceImageFilte
    * valid after the filter has executed.  Size of the background
    * (object #0) is not calculated.  */
   ObjectSizeType
-  GetSizeOfObjectInPixels(LabelType obj) const
+  GetSizeOfObjectInPixels(SizeValueType obj) const
   {
     if (obj > 0 && obj <= m_NumberOfObjects)
```

Comment:
Another solution might be to cast `obj` to `SizeValueType`, but then only when `obj` is greater than zero. In code:
```cpp
    GetSizeOfObjectInPixels(LabelType obj) const
    {
      if (obj > 0 && static_cast<SizeValueType>(obj) <= m_NumberOfObjects)
```

Doing so would avoid integer overflow when the passed argument is in fact a negative number. (I do not know if that is a concern to anyone. Just something to consider. Would people ever call something like `GetSizeOfObjectInPixels(-1)` anyway?)

Reply:
I guess this would work and has the advantage to not break the public API.  From a strict language point of view, it is a bit confusing to enable `LabelType` parameter when accessing an array-like container by index. I'll consider this solution as a patch in order to keep the public API stable.

Reply:
> If it makes no sense to use the class with signed pixel type, it should be enforced at compile-time

I agree. But for your use case, it _does_ make sense to have a signed pixel type, right?

Reply:
> > If it makes no sense to use the class with signed pixel type, it should be enforced at compile-time
> 
> I agree. But for your use case, it _does_ make sense to have a signed pixel type, right?

Yes

Reply:
Does your image also have negative pixel values? And if so, how does (or how _should_) the filter deal with them?

Reply:
For the record, the parameter type of `GetSizeOfObjectInPixels` was actually changed before, from `unsigned long` to `LabelType`, with commit 1bd69fc364bcced99bf48a3739e5acd3719b6e04 Luis Ibanez (@luisibanez), April 28, 2009: _COMP: Fixing warnings about casting._

Reply:
> Does your image also have negative pixel values? And if so, how does (or how _should_) the filter deal with them?

Sorry for my poor explanations. We use legacy code that use signed short images in a processing pipeline where images have only positive value...

Reply:
No problem! So it still appears useful to allow signed pixel types.

The compile error you reported in issue #4865 is only an _error_ because you compiled with GCC option `-Werror`, right? Otherwise I guess it would just be a warning. Anyway, we also try to avoid any warning 😇 

I would like it if you would amend (+ force-push) your commit to use `static_cast<SizeValueType>`, and restore the original API. But I leave it up to you. 👍

Reply:
Done